### PR TITLE
#118: DB cleanup and IT tests addition to CI

### PR DESCRIPTION
**Pull request by lsulak**
Preparation for #118 - I don't really want to do all the DB setup for tests manually, it's too much effort. I used Flyway to automate it. It's not ideal - the actual data insertions and cleanups after each test - but it's better than it was and we can improve it on the next increment.

What was done:
* DB files (SQL and DDL) were moved and renamed / polished to be compatible with Flyway tool; it seems like a lot of changes and files, but really it was mostly small changes
* I created `.github/workflows/integration_tests.yml` - but there are many things to improve in the future:
  * test coverage
  * better documentation perhaps (#116)
  * better test isolation - test data init and cleanup (#121)

--

Relates to #118 
Relates to #116
Relates to #121
Relates to #19

### CODE REVIEW

**Review by salamonpavel — DISMISSED** [review]
Good job, this was definitely missing. Let's address this also in other projects.

File: database/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+## About
```
Comment (benedeki) [inline database/README.md:1]: **To consider:**
I would rename the module to `test_database` or `demo_database`.

File: build.sbt [hunk 2]
```diff
@@ -48,6 +48,14 @@ lazy val commonJacocoReportSettings: JacocoReportSettings = JacocoReportSettings
 lazy val commonJacocoExcludes: Seq[String] = Seq(
 )
 
+enablePlugins(FlywayPlugin)
+flywayUrl := FlywayConfiguration.flywayUrl
+flywayUser := FlywayConfiguration.flywayUser
+flywayPassword := FlywayConfiguration.flywayPassword
+flywayLocations := FlywayConfiguration.flywayLocations
+flywaySqlMigrationSuffixes := FlywayConfiguration.flywaySqlMigrationSuffixes
+libraryDependencies ++= flywayDependencies
```
Comment (benedeki) [inline build.sbt:57]: I think it adds to the global dependencies, doesn't it? I think flyway dependencies shouldn't be a generic dependency.

File: build.sbt [hunk 3]
```diff
@@ -48,6 +48,14 @@ lazy val commonJacocoReportSettings: JacocoReportSettings = JacocoReportSettings
...
+
```
Comment (benedeki) [inline build.sbt:58]: `database` module is not present. Also isn't it overlapping the modules `database` and `examples`?

File: database/README.md (hunk 1 above)
Comment (lsulak) [reply database/README.md:1]: Okay, agreed, will do that

File: build.sbt (hunk 2 above)
Comment (lsulak) [reply build.sbt:57]: I believe this is what @salamonpavel was dealing with in the past and there was no way how to circumvent this (I know :( ) by extracting / moving elsewhere. Anyway, it's part of the build, i.e. general dependency: https://github.com/AbsaOSS/atum-service/blob/master/build.sbt#L53-L59

(remember, Flyway is temporary anyway)

File: build.sbt (hunk 3 above)
Comment (lsulak) [reply build.sbt:58]: good point, an overlook on my side about missing `database` - in terms of examples, I didn't care about them here in this PR, but why not, I'll think about deleting / merging or moving them.

Comment (lsulak) [reply build.sbt:58]: based on the content of those tests in examples/, I decided to drop them completely. All the tests there are ignored, it refers to `menas` DB, and objects that are not in this repo, and those individual functions that would be tested are basically nicely covered in actual tests in Slick.

Comment (benedeki) [reply build.sbt:58]: Good, no objections

**Review by benedeki — APPROVED** [review]
Well, except the Flyway dependency I think it's OK. That one can be addressed later.

File: build.sbt (hunk 2 above)
Comment (Zejnilovic) [reply build.sbt:57]: I agree with @benedeki 

```sbt
lazy val modulNeedingFlyaway = project
  .enablePlugins(FlywayPlugin)
  .settings(
    flywayUrl := FlywayConfiguration.flywayUrl
    flywayUser := FlywayConfiguration.flywayUser
    flywayPassword := FlywayConfiguration.flywayPassword
    flywayLocations := FlywayConfiguration.flywayLocations
    flywaySqlMigrationSuffixes := FlywayConfiguration.flywaySqlMigrationSuffixes
    libraryDependencies ++= flywayDependencies
  )
```

and then run it like `sbt modulNeedingFlyaway/flywayMigrate`

Comment (benedeki) [reply build.sbt:57]: Can we redefine the `flywayMigrate` to behave like `sbt modulNeedingFlyaway/flywayMigrate`?

Comment (Zejnilovic) [reply build.sbt:57]: Like this `addCommandAlias("flywayMigrate", "modulNeedingFlyaway/flywayMigrate")`?

Comment (lsulak) [reply build.sbt:57]: Let me try, thanks @Zejnilovic. To be fair I haven't been playing with this one as my primary objective here was to actually have a IT suite embedded in our CI - but I agree that we can improve this mechanism in build.sbt

Comment (benedeki) [reply build.sbt:57]: That's why it got approved, not blocking the whole thing 😉

**Pull request merged**